### PR TITLE
Group1

### PR DESCRIPTION
**Pull request by temari**

### CODE REVIEW

File: Java/src/main/java/com/gildedrose/Item.java [hunk 1]
```diff
@@ -2,20 +2,53 @@
 
 public class Item {
 
-    public String name;
+    private ItemTypeEnum itemType;
 
     public int sellIn;
```
Comment (kufuntu) [inline Java/src/main/java/com/gildedrose/Item.java:7]: this should also be private

File: Java/src/main/java/com/gildedrose/GildedRose.java [hunk 2]
```diff
@@ -8,54 +8,43 @@ public GildedRose(Item[] items) {
...
-                            }
-                        }
-                    } else {
-                        items[i].quality = items[i].quality - items[i].quality;
-                    }
-                } else {
-                    if (items[i].quality < 50) {
-                        items[i].quality = items[i].quality + 1;
+                case DEXTERITY_VEST:
+                case MONGOOSE:
```
Comment (kufuntu) [inline Java/src/main/java/com/gildedrose/GildedRose.java:41]: consider changing that to default, or you need to add error handling e.g. throw an exception in the enum creator

File: Java/src/main/java/com/gildedrose/GildedRose.java [hunk 3]
```diff
@@ -8,54 +8,43 @@ public GildedRose(Item[] items) {
...
-                    if (items[i].name.equals("Backstage passes to a TAFKAL80ETC concert")) {
-                        if (items[i].sellIn < 11) {
-                            if (items[i].quality < 50) {
-                                items[i].quality = items[i].quality + 1;
-                            }
-                        }
+        for (Item currItem : items) {
+            switch (currItem.getItemType()) {
+                case AGED_BRIE:
+                    currItem.checkQualityAndIncrease(1);
```
Comment (kufuntu) [inline Java/src/main/java/com/gildedrose/GildedRose.java:15]: consider moving that logic into seperate methods its a quite long logic block

File: Java/src/main/java/com/gildedrose/ItemTypeEnum.java [hunk 4]
```diff
@@ -0,0 +1,29 @@
+package com.gildedrose;
+
+public enum ItemTypeEnum {
```
Comment (DGCAS20) [inline Java/src/main/java/com/gildedrose/ItemTypeEnum.java:3]: Enum finden wir ganz gute Lösung.

**Review by DGCAS20 — COMMENTED** [review]
Good job

File: Java/src/main/java/com/gildedrose/Item.java [hunk 5]
```diff
@@ -2,20 +2,53 @@
 
 public class Item {
 
-    public String name;
+    private ItemTypeEnum itemType;
```
Comment (judith600) [inline Java/src/main/java/com/gildedrose/Item.java:5]: changed item class

File: Java/src/main/java/com/gildedrose/GildedRose.java (hunk 2 above)
Comment (judith600) [reply Java/src/main/java/com/gildedrose/GildedRose.java:41]: 👍

File: Java/src/main/java/com/gildedrose/GildedRose.java (hunk 3 above)
Comment (judith600) [reply Java/src/main/java/com/gildedrose/GildedRose.java:15]: 👍

**Pull request merged**